powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/SignatureUtils.java

```diff
@@ -83,46 +85,118 @@ public boolean validateECDSASignature(byte[] signedBytes, byte[] signature, Publ
     }
 
     /**
-     * Compute PowerAuth signature for given data using a secret signature keys and counter byte array.
+     * Compute decimal formatted PowerAuth signature for given data using a secret signature keys and counter byte array.
      *
      * @param data Data to be signed.
      * @param signatureKeys Keys for computing the signature.
      * @param ctrData Counter byte array / derived key index.
-     * @return PowerAuth signature for given data.
+     * @return Decimal formatted PowerAuth signature for given data.
+     * @throws GenericCryptoException In case signature computation fails.
+     * @throws CryptoProviderException In case cryptography provider is incorrectly initialized.
+     */
+    private String computePowerAuthDecimalSignature(byte[] data, List<SecretKey> signatureKeys, byte[] ctrData) throws GenericCryptoException, CryptoProviderException {
+        // Prepare holder for signature components
+        final String[] signatureStringComponents = new String[signatureKeys.size()];
+        // Compute signature components
+        final byte[][] signatureComponents = computePowerAuthSignatureComponents(data, signatureKeys, ctrData);
```

Comment:
A style comment - for me `byte[][]` is a bit confusing and smells like C. I would personally prefer `List<byte[]>`.

Reply:
You're right. It was also for me a little bit confusing, but did not found a right replacement.